docs/versioning.md

```diff
@@ -0,0 +1,101 @@
+* An ONNX ModelProto that represents a given graph - that is, the contents of a model. We refer to this version as the *model version* and it is represented by the `ModelProto.model_version` field.    
+
+The versioning of all three of these entity types is distinct, and largely independent. That is, ONNX versions the IR format at a different rate than the operators defined by ONNX - the former will version much slower than the latter. 
+
+While the versioning mechanisms are clearly specified in this document, the policies for version management are only mandated for IR version and operator verison. 
+
+For model version, ONNX users and systems MAY follow whatever local customs make sense, however, to facilitate shared libraries or repositories of ONNX models to be managable, models SHOULD adhere to the policies described under Model versioning 
+
+
+
+### SemVer, Files and Frameworks
+
+ONNX builds on the principles and syntax defined by [SemVer 2.0.0](http://semver.org/spec/v2.0.0.html). Throughout this document, we use the terms *breaking change*, *non-breaking change*, and *patch* consistent with SemVer 2.0.0. 
+
+The SemVer specification is written in terms of the dependencies between the caller and callee of a given API.  With respect to operator definitions, the NodeProto represents the caller, and the operator specification represents the callee.
+
+Because ONNX models are serialized files, not APIs, it's worth making clear how the dependency between a serialized model and a piece of software that consumes that model relate.  As a rough approximation, the serialized model plays the role of an API's *callee*; the consumer of the seriaized model plays the role of the API's *caller.*
+
+The ONNX versioning principles are based on [Postel's law](https://en.wikipedia.org/wiki/Robustness_principle): be conservative in what you do, be liberal in what you accept from others.  
+
+1. A producer of a given ONNX model (and the ONNX specification itself) MUST strictly adhere to the rules for breaking vs. non-breaking changes defined in this specification.
+2. A consumer of a given ONNX model SHOULD consume an updated ONNX file provided there are no breaking changes in the new ONNX file's IR version, referenced operator versions, or model version (e.g., the MAJOR version numbers are have not changed between the two ONNX files) .
+3. A consumer of a given ONNX model MAY consume an updated ONNX file provided there are one or more breaking changes in the new ONNX file's IR version, referenced operator versions, or model version.
+
+Per the rules of SemVer 2.0, during the initial development of ONNX, we use a MAJOR version of 0 for both the IR version and operator version and will only increment the MINOR version in the face of either a breaking change or the need to stabiize a version for specific engineering purposes. [ISSUE: decide how we will stabilize and archive released versions of ONNX] 
```

Comment:
typo? "and will only increment the MINOR version" => "MAJOR version"?

Reply:
Not a typo.  It only applies to the pre-1.0.0 policy, and we need a way to snap releases and still patch them.

Reply:
And I added text to make that somewhat clearer.

Reply:
Thank you! Understood.